packages/db/index.js

```diff
@@ -1274,4 +1274,52 @@ export class DBClient {
       _id: data.id
     }
   }
+
+  /**
+   * Set the customerId that corresponds to a particular userId.
+   * The Customer ID may be a stripe.com customer id (not a foreign key within this db)
+   * @param {string} userId
+   * @param {string} customerId
+   */
+  async upsertUserCustomer (userId, customerId) {
+    const { data, error } = await this._client
+      .from('user_customer')
+      .upsert(
+        {
+          user_id: userId,
+          customer_id: customerId
+        },
+        {
+          onConflict: 'user_id'
+        }
+      )
+      .single()
+    if (error) {
+      throw new DBError(error)
+    }
+    return {
+      _id: data.id
+    }
+  }
+
+  /**
+   * Get the Customer for a user
+   * @param {string} userId
+   * @returns {null|{ id: string }} customer
+   */
+  async getUserCustomer (userId) {
+    const { data, error } = await this._client
+      .from('user_customer')
+      .select(['customer_id'].join(','))
```

Comment:
No need for the join

Reply:
according to the docs (and tsc types, which aren't being checked in db package yet), it must be a string. https://supabase.github.io/postgrest-js/classes/PostgrestQueryBuilder.html#select

(not disagreeing that the join works now at runtime, but I'd prefer to keep it here so it doesn't have to change again if/when we enable type checking https://github.com/web3-storage/web3.storage/issues/1860 )